app/register/page.tsx

```diff
@@ -1,87 +1,139 @@
-import Link from 'next/link';
-import { redirect } from 'next/navigation';
-import { createUser, getUser } from '@/lib/db';
-import { SimpleLayout } from '@/components/page-layouts';
-import { Label } from '@/components/ui/label';
-import { Input } from '@/components/ui/input';
-import { Button } from '@/components/ui/button';
+"use client";
 
-export default function Page() {
-
-  async function register(formData: FormData) {
-    'use server';
-    let name = formData.get('name') as string;
-    let phone = formData.get('phone') as string;
-    let email = formData.get('email') as string;
-    let password = formData.get('password') as string;
-    let confirmPassword = formData.get('confirm-password') as string;
-    
-    // TODO do confirm password checking locally with react-hook-forms
-    if (confirmPassword !== password) {
-      return 'Passwords do not match';
-    }
+import Link from "next/link";
+import { redirect } from "next/navigation";
+import { registerUser } from "@/lib/actions";
+import { SimpleLayout } from "@/components/page-layouts";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { RegisterSchema, RegisterSchemaType } from "@/lib/types";
 
-    let user = await getUser(email);
+export default function Page() {
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors },
+  } = useForm<RegisterSchemaType>({
+    resolver: zodResolver(RegisterSchema),
+  });
 
-    if (user.length > 0) {
-      // TODO: Handle errors with useFormStatus
-      return 'User already exists';
-    } else {
-      await createUser(name, phone, email, password);
-      redirect('/login');
+  const onSubmit: SubmitHandler<RegisterSchemaType> = async (data) => {
+    try {
+      await registerUser(data);
+      redirect("/login");
+    } catch (error) {
+      console.error("Error:", error);
```

Comment:
We can leave this for another PR, but I think we can add better error handling here, and show an error state in the UI